### DISCUSSION TITLE
Have the MIDI in/out of the panel in "parallel" to the DAW

### DISCUSSION THREAD

**Discussion by dervur** [RomanKubiak/ctrlr#671] [Q&A] [answered]
Hello there, is there a way to have the MIDI in/out settings go in parallel with the DAW rather than in an exclusive way? I tried to search in the forum but probably I'm calling things with the wrong name, so I apologize if it has already been asked.

What I mean is, is there a way to have the exported panel have the ability to select MIDI in/out without issues because the DAW is already using them? Often there is an error message saying that you can't use that MIDI port. I think this would make things not make sense a lot in the real world as people might use one MIDI out for 16 devices, one for each channel, and it is not reasonable that each exported VST occupies one channel and that it can't be selected by other VSTs...
This would save from this kind of issues:
![image](https://github.com/user-attachments/assets/6417c79d-8b41-45e7-a7fe-03724aed6418)

**Comment by damiensellier** [comment, marked as answer]
It's a OS level problem. you're midi port is already in use.
If you're using WIN10 and previous versions, most of the midi interface have just one single access to the midi port. 
macOS, on the other side, is multi client by default so, multiple apps can use the same midi port simultaneously. the OS will merge the messages.

WIN 11 has a new midi implementation and allows multi client, but the drivers have to follow the new standards.

So, you can use the midi thru capabilities if you're inside a DAW with a VST.

Or if it's in another situation you need to create a virtual midi port, assign your hardware  to this virtual midi port and use a virtual midi patchbay to split to the different midi devices to make a single hardware midi port multiclient.

https://www.soundquest.org/Help/MidiQuest12/using_plugins_with_single_clie.htm

https://www.tobias-erichsen.de/software.html

https://soundigy.com/midipatchbay.php

It works great on WIN10 I do as a workaround to transform midi messages with a ctrlr panel.

Hope it helps.

Damien


https://soundigy.com/midipatchbay.php

**Reply by dervur** [reply]
Thank you very much. I guess this should solve everything. Maybe I should export a version of the VST as an effect and not only instrument as it is doing now... In that way I don't need to touch any MIDI settings within the panel itself but I would from the MIDI track.

**Reply by damiensellier** [reply]
This is what I do for panel which controls external FX. The ctrlr panel sits as an insert plugin on the AUX Channel in Cubase, it makes more sense than having it in the instruments rack.

**Reply by dervur** [reply]
I guess it can also work for instruments by putting them on a MIDI track and selecting in/out on the midi track rather than the VST/panel. In order to export it as a VST effect I need to compile by deselecting everything except MIDI effect plugin in projucer, right? Or I have to deselect only plugin is a synth? Or do I need to do something in Visual Studio?
![image](https://github.com/user-attachments/assets/f1750d89-0938-4d13-80f5-177f9ce3686e)

**Comment by damiensellier** [comment]
You can just remove Plugin is is a Synth. I think if I remember well, if you export as VST3 you won't need "Midi Effect Plugin" because the VST3 standards allows midi in/out anyway.
Ctrlr will be seen as a VST3 audio plugin that you can insert on any audio channel in your DAW. It's perfect to control a reverb or FX hardware like a plugin as insert on your audio or AUX channel.
Never tried to disable Plugin is a Synth as standalone app though. I can't tell if it's going to work properly in this case.

**Reply by dervur** [reply]
Well I'll try in the future and see what happens. I'll write here the result.
